src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/ActiveDirectoryAuthenticationProvider.cs

```diff
@@ -119,26 +119,41 @@ public override async Task<SqlAuthenticationToken> AcquireTokenAsync(SqlAuthenti
             string scope = parameters.Resource.EndsWith(s_defaultScopeSuffix) ? parameters.Resource : parameters.Resource + s_defaultScopeSuffix;
             string[] scopes = new string[] { scope };
 
+            string authority = parameters.Authority;
             int seperatorIndex = parameters.Authority.LastIndexOf('/');
-            string tenantId = parameters.Authority.Substring(seperatorIndex + 1);
-            string authority = parameters.Authority.Remove(seperatorIndex + 1);
 
-            TokenRequestContext tokenRequestContext = new TokenRequestContext(scopes);
+            // Authority Url does not always contain "<tenantId>", e.g. for Kusto cluster dbs
+            // Optionally split Tenant Id in such cases.
+            if (Guid.TryParse(parameters.Authority.Substring(seperatorIndex + 1), out Guid tenantId))
```

Comment:
The `tenantId` isn't always a GUID, can be `"microsoft.com"`, `"ame.gbl"`, ... or `"common"`. In the latter case token will be issued by the home tenant of the user that you signin with.
Also could resolve it under `if (parameters.AuthenticationMethod == SqlAuthenticationMethod.ActiveDirectoryDefault)` since SPA auth uses the full authority URL now.

Reply:
After reading through [documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-client-application-configuration#how-to-specify-the-audience-in-your-codeconfiguration), it seems to work fine either way... we need not change this part and can continue as-is. The term `tenantId` was misleading and our thought was that it's only applicable to `Guid` values. It makes sense to call it `audience` instead. I've updated code accordingly.